src/components/study-pane.js

```diff
@@ -1313,505 +188,130 @@ const StudyPane = (props) => {
-            } else if (networkLoadingFail) {
-                setErrorMsgId('networkLoadingFail');
-            } else if (geoDataLoadingFail) {
-                setErrorMsgId('geoDataLoadingFail');
-            }
-        }
-    }, [
-        studyNotFound,
-        geoDataLoadingFail,
-        networkLoadingFail,
-        studyExistsChecked,
-        waitingLoadGeoData,
-        isNetworkLoadingDone,
-    ]);
-
-    if (errorMsgId) {
-        return (
-            <PageNotFound
-                message={
-                    <FormattedMessage
-                        id={errorMsgId}
-                        values={{ studyUuid: studyUuid }}
-                    />
-                }
-            />
-        );
-    } else {
-        return (
-            <>
-                {isWaitingLoading() && (
-                    <LoaderWithOverlay
-                        color="inherit"
-                        loaderSize={70}
-                        isFixed={true}
-                        loadingMessageText={
-                            waitingLoadGeoData
-                                ? 'loadingGeoData'
-                                : 'loadingReport'
-                        }
-                    />
-                )}
-                {/*Rendering the map is slow, do it once and keep it display:none*/}
-                <div
-                    className="singlestretch-parent singlestretch-child"
-                    style={{
-                        display: props.view === StudyView.MAP ? null : 'none',
-                    }}
-                >
-                    {renderMapView()}
-                </div>
-                <div
-                    className="singlestretch-parent singlestretch-child"
-                    style={{
-                        display:
-                            props.view === StudyView.SPREADSHEET
-                                ? null
-                                : 'none',
-                    }}
-                >
-                    {renderTableView()}
-                </div>
-                <div
-                    className="singlestretch-parent singlestretch-child"
-                    style={{
-                        display:
-                            props.view === StudyView.RESULTS ? null : 'none',
-                    }}
-                >
-                    {renderResultsView()}
-                </div>
-                <div
-                    className="singlestretch-parent singlestretch-child"
-                    style={{
-                        display: props.view === StudyView.LOGS ? null : 'none',
-                    }}
-                >
-                    {renderLogsView()}
-                </div>
-            </>
-        );
-    }
+            </div>
+            <div
+                className="singlestretch-parent singlestretch-child"
+                style={{
+                    display: props.view === StudyView.LOGS ? null : 'none',
```

Comment:
Then every tab is rendered but with a style.display to null or non depending of the props.view value ? is it optimized ?

Reply:
not mine, but for the map it is used to keep all the rendering

Reply:
Ok I understand for the geographical map...